app/routes/farmer-apply/confirmation.js

```diff
@@ -13,18 +13,21 @@ module.exports = {
       // TODO: Get this data based on eligibility or the applicant
       // const { sbi } = request.payload
       const organisation = session.getOrganisation(request)
-
       // TODO: should the reference number be a particular format?
-      const reference = uuid().split('-').shift().toLocaleUpperCase('en-GB')
+      const reference = uuidv4().split('-').shift().toLocaleUpperCase('en-GB')
+      session.setApplication(request, 'applicationId', reference)
+      session.setApplication(request, 'sessionId', request.yar.id)
+      session.setApplication(request, 'organisation', organisation)
 
-      // TODO: Check an email hasn't been sent already and store the fact that this has been sent
-      const result = await sendEmail(templateIdApplicationComplete, organisation.email, { personalisation: { name: organisation.name, reference }, reference })
+      const application = session.getApplication(request)
+      sendMessage(application, applicationRequestMsgType, applicationRequestQueue, { sessionId: request.yar.id })
+      const response = await receiveMessage(request.yar.id, applicationResponseQueue)
 
-      if (!result) {
-        return boom.internal()
+      if (response) {
+        console.info('Response received:', util.inspect(response, false, null, true))
```

Comment:
I think there needs to be some error handling here. Rather than logging the response, can the object be checked to see if there was a valid response and if not log an error with `boom` e.g.
```js
if (!response.valid) {
    return boom.internal()
}
```